io/net/infiniband/ib_bw_perf.py

```diff
@@ -0,0 +1,111 @@
+from avocado.utils import distro
+
+
+class Bw_Perf(Test):
+    '''
+    check the availability of perftest package installed
+    perftest package should be installed
+    '''
+    def setUp(self):
+        '''
+        To check and install dependencies for the test
+        '''
+        sm = SoftwareManager()
+        detect_distro = distro.detect()
+        depends = ["openssh-clients"]
+        if detect_distro.name == "Ubuntu" or detect_distro.name == "redhat":
+            depends.append("perftest")
+        for package in depends:
+            if not sm.check_installed(package):
+                self.skip("%s package is need to test" % package)
+        interfaces = netifaces.interfaces()
+        self.flag = self.params.get("ext_flag", default="0")
+        self.IF = self.params.get("Iface", default="")
+        self.PEER_IP = self.params.get("PEERIP", default="")
+        if self.IF not in interfaces:
+            self.skip("%s interface is not available" % self.IF)
+        if self.PEER_IP == "":
+            self.skip("%s peer machine is not available" % self.PEER_IP)
+        self.CA = self.params.get("CA_NAME", default="mlx4_0")
+        self.PORT = int(self.params.get("PORT_NUM", default="1"))
+        self.PEER_CA = self.params.get("PEERCA", default="mlx4_0")
+        self.PEER_PORT = int(self.params.get("PEERPORT", default="1"))
+        self.to = self.params.get("timeout", default="600")
+
+    def bwperf_exec(self, arg1, arg2, arg3):
+        '''
+        bandwidth performance exec function
+        '''
+        logs = "> /tmp/ib_log 2>&1 &"
+        msg = "timeout %s %s -d %s -i %d %s %s %s" \
+            % (self.to, arg1, self.PEER_CA, self.PEER_PORT,
+                arg2, arg3, logs)
+        cmd = "ssh %s %s" % (self.PEER_IP, msg)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.fail("ssh failed to remote machine")
+        time.sleep(2)
+        self.log.info("client data for %s(%s)" % (arg1, arg2))
+        self.log.info("%s -d %s %s -i %d %s %s"
+                      % (arg1, self.CA, self.PEER_IP, self.PORT,
+                         arg2, arg3))
+        tmp = "timeout %s %s -d %s -i %d %s %s %s" \
+            % (self.to, arg1, self.CA, self.PORT, self.PEER_IP,
+                arg2, arg3)
+        if process.system(tmp, shell=True, ignore_status=True) != 0:
+            self.fail("test failed")
+        self.log.info("server data for %s(%s)" % (arg1, arg2))
+        self.log.info("%s -d %s -i %d %s %s"
+                      % (arg1, self.PEER_CA, self.PEER_PORT,
+                         arg2, arg3))
+        msg = "timeout %s cat /tmp/ib_log; rm -rf /tmp/ib_log" % self.to
+        cmd = "ssh %s %s" % (self.PEER_IP, msg)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.fail("test failed")
+
+    def test_ib_bandwidth(self):
+        '''
+        test options are mandatory
+        ext test options are depends upon user
+        '''
```

Comment:
I don't follow the meaning of this comment is this indicate definition of function or restriction ?